pants-plugins/uses_services/mongo_rules_test.py

```diff
@@ -0,0 +1,170 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import pytest
+
+from pants.engine.internals.scheduler import ExecutionError
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+from .exceptions import ServiceMissingError
+from .mongo_rules import (
+    MongoIsRunning,
+    UsesMongoRequest,
+    rules as mongo_rules,
+)
+from .platform_rules import Platform
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *mongo_rules(),
+            QueryRule(MongoIsRunning, (UsesMongoRequest, Platform)),
+        ],
+        target_types=[],
+    )
+
+
+def run_mongo_is_running(
+    rule_runner: RuleRunner,
+    uses_mongo_request: UsesMongoRequest,
+    mock_platform: Platform,
+    *,
+    extra_args: list[str] | None = None,
+) -> MongoIsRunning:
+    rule_runner.set_options(
+        [
+            "--backend-packages=uses_services",
+            *(extra_args or ()),
+        ],
+        env_inherit={"PATH", "PYENV_ROOT", "HOME"},
+    )
+    result = rule_runner.request(
+        MongoIsRunning,
+        [uses_mongo_request, mock_platform],
+    )
+    return result
+
+
+def platform(
+    arch="",
+    os="",
+    distro="",
+    distro_name="",
+    distro_codename="",
+    distro_like="",
+    distro_major_version="",
+    distro_version="",
+    mac_release="",
+    win_release="",
+) -> Platform:
+    """Create a Platform with all values defaulted to the empty string."""
+    return Platform(
+        arch=arch,
+        os=os,
+        distro=distro,
+        distro_name=distro_name,
+        distro_codename=distro_codename,
+        distro_like=distro_like,
+        distro_major_version=distro_major_version,
+        distro_version=distro_version,
+        mac_release=mac_release,
+        win_release=win_release,
+    )
+
+
+# Warning this requires that mongo be running
+def test_mongo_is_running(rule_runner: RuleRunner) -> None:
+    request = UsesMongoRequest()
+    mock_platform = platform()
+
+    # we are asserting that this does not raise an exception
+    is_running = run_mongo_is_running(rule_runner, request, mock_platform)
+    assert is_running
+
+
+@pytest.mark.parametrize(
+    "mock_platform",
+    (
+        platform(),  # empty
+        # platform(
+        #    arch="x86_64",
+        #    os="Linux",
+        #    distro="",
+        #    distro_name="",
+        #    distro_codename="",
+        #    distro_like="",
+        #    distro_major_version="",
+        #    distro_version="",
+        # ),
```

Comment:
I need more samples. I want to grab at least one from each of our target platforms.